### PR TITLE
switch the host persistence to correctly load compatibility fields

### PR DESCRIPTION
**Pull request by DavidVorick**
There were some incorrectly named fields in the pre-verion 1.0 releases
of the host. The names have since been corrected, but compatibility code
needed to be written to make sure that old hosts doing an upgrade would
not experience any issues due to the renaming.

### CODE REVIEW

**Comment by DavidVorick** [comment]
Please double check the spelling for me.

File: modules/host/persist.go [hunk 1]
```diff
@@ -171,6 +171,44 @@ func (h *Host) load() error {
...
+		FinancialMetrics struct {
+			PotentialStorageRevenue types.Currency `json:"potentialerevenue"`
+		}
+		Settings struct {
+			MinContractPrice          types.Currency `json:"contractprice"`
+			MinDownloadBandwidthPrice types.Currency `json:"minimumdownloadbandwidthprice"`
+			MinStoragePrice           types.Currency `json:"storageprice"`
+			MinUploadBandwidthPrice   types.Currency `json:"minimumuploadbandwidthprice"`
+		}
+	}
```
Comment (DavidVorick) [inline modules/host/persist.go]: These values don't reflect what is in master branch, but they do reflect what is in the v0.6.0 release.

Comment (DavidVorick) [inline modules/host/persist.go]: I used git blame to find other fields where compatibility had been accidentally been broken. I'm not sure the best way to be 100% sure that others weren't missed.

I only checked the host fields, not fields from other modules.

There may be some chaos this upgrade.

Comment (lukechampine) [inline modules/host/persist.go]: shouldn't this be `COMPAT v0.6.0`?

File: modules/host/persist.go [hunk 2]
```diff
@@ -171,6 +171,44 @@ func (h *Host) load() error {
...
+	err = h.dependencies.loadFile(persistMetadata, &compatPersistence, filepath.Join(h.persistDir, settingsFile))
+	if err != nil {
+		return err
+	}
+	// Load the compat values, but only if the compat values are non-zero and
+	// the real values are zero.
+	if compatPersistence.FinancialMetrics.PotentialStorageRevenue.Cmp(types.ZeroCurrency) > 0 && p.FinancialMetrics.PotentialStorageRevenue.Cmp(types.ZeroCurrency) == 0 {
```
Comment (lukechampine) [inline modules/host/persist.go]: use `IsZero` to make these more readable and concise

File: modules/host/persist.go (hunk 1 above)
Comment (DavidVorick) [inline modules/host/persist.go]: I thought the convention was to label the COMPAT for the version where the change was introduced, to give us an idea of how long people have had to upgrade away from the change.

File: modules/host/persist.go (hunk 2 above)
Comment (DavidVorick) [inline modules/host/persist.go]: forgot we had that function.

**Comment by DavidVorick** [comment]
pushed into #1254

**Pull request closed without merging**